python/lsst/obs/decam/crosstalk.py

```diff
@@ -119,128 +118,46 @@ def prepCrosstalk(self, dataRef):
+        if crosstalk.interChip is None or len(crosstalk.interChip) == 0:
+            return None
+
         # Retrieve visit and ccdnum from 'victim' so we can look up 'sources'
-        victim_exposure = dataRef.get()
-        det = victim_exposure.getDetector()
         visit = dataRef.dataId['visit']
-        ccdnum = det.getId()
-        ccdnum_str = '%02d' % ccdnum
         crosstalkSources = defaultdict(list)
+
         decamisr = IsrTask()  # needed for source_exposure overscan correction
-        for amp in det:
-            victim = ccdnum_str + amp.getName()
-            for source in sources[victim]:
-                source_ccdnum = int(source[:2])
-                try:
-                    source_exposure = butler.get('raw', dataId={'visit': visit, 'ccdnum': source_ccdnum})
-                except RuntimeError:
-                    self.log.warn('Cannot access source %d, SKIPPING IT' % source_ccdnum)
-                else:
-                    self.log.info('Correcting victim %s from crosstalk source %s' % (victim, source))
-                    if 'A' in source:
-                        amp_idx = 0
-                    elif 'B' in source:
-                        amp_idx = 1
-                    else:
-                        self.log.fatal('DECam source amp name does not contain A or B, cannot proceed')
-                    source_amp = source_exposure.getDetector()[amp_idx]
-                    # If doCrosstalkBeforeAssemble=True, then use getRawBBox().  Otherwise, getRawDataBBox().
-                    source_dataBBox = source_amp.getRawBBox()
-                    # Check to see if source overscan has been corrected, and if not, correct it first
-                    if not source_exposure.getMetadata().exists('OVERSCAN'):
-                        decamisr.overscanCorrection(source_exposure, source_amp)
-                        self.log.info('Correcting source %s overscan before using to correct crosstalk'
-                                      % source)
-                    source_image = source_exposure.getMaskedImage().getImage()
-
-                    source_data = source_image.Factory(source_image, source_dataBBox, deep=True)
-                    crosstalkSources[victim].append((source_data, coeffs[(victim, source)], amp_idx))
+        for sourceDet in crosstalk.interChip:
+            try:
+                source_exposure = butler.get('raw', dataId={'visit': visit, 'ccd': sourceDet})
+            except RuntimeError:
+                self.log.warn('Cannot access source %s, SKIPPING IT' % sourceDet)
+            else:
+                self.log.info('Correcting victim %s from crosstalk source %s' %
+                              (crosstalk._detectorName, sourceDet))
+
+                for amp in source_exposure.getDetector():
+                    decamisr.overscanCorrection(source_exposure, amp)
+
+                # Do not assemble here, as DECam CT is corrected before assembly.
```

Comment:
Is this always the case, now? I remember the old `doCrosstalkBeforeAssemble` but I don't remember when it was True or False or why.

Reply:
HSC and DECam have different conventions, and I think there's a linearity correction term that changes the pixel values differently depending on when it's applied.

Reply:
Fair enough, but there is still a `doCrosstalkBeforeAssemble` in the ip_isr config. It defaults to False in ip_isr, but unless that config is going away, I would prefer to refer to it here (even if it hardwired one way or the other and there is virtually no reason to have it set differently).